foxpuppet/windows.py

```diff
@@ -0,0 +1,48 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from contextlib import contextmanager
+
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.wait import WebDriverWait
+
+
+class Windows(object):
+
+    def __init__(self, selenium):
+        self.selenium = selenium
+        self._handle = None
+
+    @property
+    def all(self):
+        return self.selenium.window_handles
+
+    @property
+    def current(self):
+        return self.selenium.current_window_handle
+
+    def focus(self, handle):
+        return self.selenium.switch_to.window(handle)
+
+    def get_new_window_handle(self, handles):
+        """Get new window handle"""
+        if handles != self.selenium.window_handles:
+            new_handle = self.selenium.window_handles[len(handles):]
+        return new_handle
+
+    @contextmanager
+    def wait_for_new_window(self):
+        handles_before = self.selenium.window_handles
+        yield
+        WebDriverWait(self.selenium, 10).until(
+            lambda driver: len(handles_before) != len(driver.window_handles))
```

Comment:
I wonder if it would make more sense to register an event listener before we trigger the window open, and perhaps the listener would have a reference to the window handle, which we could use? Do you have any suggestions @whimboo?

Looking at Puppeteer, there's some code we could reuse [here](https://dxr.mozilla.org/mozilla-central/source/testing/puppeteer/firefox/firefox_puppeteer/ui/windows.py#343-381). It looks like we're blocked on [bug 1121698](https://bugzilla.mozilla.org/show_bug.cgi?id=1121698) for making this more robust.